lib/runner.js

```diff
@@ -662,6 +662,11 @@ internals.protect = function (item, state) {
 
                 try {
                     await item.fn.call(null, flags);
+                    // wait for any pending ticks started by the test
+                    await new Promise((next) => {
+
+                        process.nextTick(next);
```

Comment:
This will work better with an `setImmediate()`, as it allows all ticks and promises to settle.